dask/dataframe/multi.py

```diff
@@ -399,11 +399,6 @@ def hash_join(
     _rhs_meta = rhs._meta_nonempty if len(rhs.columns) else rhs._meta
     meta = _lhs_meta.merge(_rhs_meta, **kwargs)
 
-    if isinstance(left_on, list):
-        left_on = (list, tuple(left_on))
-    if isinstance(right_on, list):
-        right_on = (list, tuple(right_on))
-
```

Comment:
Just wanted to double check -- we're putting `left_on` / `right_on` in `kwargs` (which is used) a few lines above, is it possible this validation should still be around but higher up in this function? If not, removing is fine by me

Reply:
It seems unlikely that we don't have test coverage for this, but nothing is failing so I'd lean to just removing